Detectors/FIT/simulation/src/Detector.cxx

```diff
@@ -0,0 +1,550 @@
+	  //	fIshunt = 2;
+	  AddHit(x,y,z, time, enDep, trackID, detID);
+	  
+	}
+      }
+    return kTRUE;
+  }
+}
+
+HitType* Detector::AddHit(float x, float y, float z, float time, float energy, Int_t trackId, Int_t detId)
+{
+  TClonesArray& clref = *mHitCollection;
+
+  Int_t size = clref.GetEntriesFast();
+
+  HitType* hit = new (clref[size]) HitType(x, y, z, time, energy, trackId, detId);
+
+}
+
+void Detector::Register() {
+  FairRootManager::Instance()->Register("FITHit", "FIT", mHitCollection, kTRUE);
+}
+
+TClonesArray* Detector::GetCollection(Int_t iColl) const
+{
+  if (iColl == 0)
+    return mHitCollection;
+  return nullptr;
+}
+
+void Detector::Reset() {}
+
+
+void Detector::CreateMaterials()
+{
+
+  Int_t isxfld = 2;     // magneticField->Integ();
+  Float_t sxmgmx = 10.; // magneticField->Max();
+
+  //   Float_t a,z,d,radl,absl,buf[1];
+  // Int_t nbuf;
+  // AIR
+  
+  Float_t aAir[4]={12.0107,14.0067,15.9994,39.948};
+  Float_t zAir[4]={6.,7.,8.,18.};
+  Float_t wAir[4]={0.000124,0.755267,0.231781,0.012827};
+  Float_t dAir = 1.20479E-3;
+  Float_t dAir1 = 1.20479E-11;
+  // Radiator  glass SiO2
+  Float_t aglass[2]={28.0855,15.9994};
+  Float_t zglass[2]={14.,8.};
+  Float_t wglass[2]={1.,2.};
+  Float_t dglass=2.65;
+  // MCP glass SiO2
+  Float_t dglass_mcp=1.3;
+  //*** Definition Of avaible FIT materials ***
+  Mixture(1, "Vacuum$", aAir, zAir, dAir1,4,wAir);
+  Mixture(2, "Air$", aAir, zAir, dAir,4,wAir);
+  Mixture( 4, "MCP glass   $",aglass,zglass,dglass_mcp,-2,wglass);
+  Mixture( 24, "Radiator Optical glass$",aglass,zglass,dglass,-2,wglass);
+  Material(11, "Aliminium$", 26.98, 13.0, 2.7, 8.9,999); 
+ 
+  Medium(1, "Air$", 2, 0, isxfld, sxmgmx, 10., .1, 1., .003, .003);
+  Medium(3, "Vacuum$", 1, 0, isxfld, sxmgmx, 10., .01, .1, .003, .003);
+  Medium(6, "Glass$", 4, 0, isxfld, sxmgmx, 10., .01, .1, .003, .003);
+  
+  Medium(7, "OpAir$", 2, 0, isxfld, sxmgmx, 10., .1, 1., .003, .003);
+  
+  Medium(15, "Aluminium$", 11, 0, isxfld, sxmgmx, 10., .01, 1., .003, .003);  
+  Medium(16, "OpticalGlass$", 24, 1, isxfld, sxmgmx, 10., .01, .1, .003, .003);
+  Medium(19, "OpticalGlassCathode$", 24, 1, isxfld, sxmgmx, 10., .01, .1, .003, .003);
+  Medium(22, "SensAir$", 2, 1, isxfld, sxmgmx, 10., .1, 1., .003, .003);
+   
+ 
+}
+
+//-------------------------------------------------------------------
+void Detector::DefineOpticalProperties()
+{
+
+// Path of the optical properties input file
+  TString inputDir;
+  const char* aliceO2env=std::getenv("O2_ROOT");
+  if (aliceO2env) inputDir=aliceO2env;
+  inputDir+="/share/Detectors/FIT/files/";
+
+  TString optPropPath =inputDir + "quartzOptProperties.txt";
+  optPropPath = gSystem->ExpandPathName(optPropPath.Data()); // Expand $(ALICE_ROOT) into real system path
+
+// Prepare pointers for arrays read from the input file
+  Float_t *aPckov=nullptr;
+  Double_t *dPckov=nullptr;
+  Float_t *aAbsSiO2=nullptr;
+  Float_t *rindexSiO2=nullptr;
+  Float_t *qeff = nullptr;
```

Comment:
Consider making these parameters `std::vector<>` and possibly adding them as members to the detector. Like this we avoid complicated allocation/deallocation and can pass them around more easily. The ownsership will also be much clearer (in particular we would like to avoid any manual `delete []` call).